src/store/filestore.rs

```diff
@@ -1,213 +1,216 @@
 use crate::{
-    error::{FileExists, FileHasNoParent, NoAnnotationFound, NoRegexMatch},
-    model::{from_yaml, to_yaml, Pod},
-    store::Store,
+    error::{FileExists, FileHasNoParent, NoAnnotationFound},
+    model::{from_yaml, to_yaml, Annotation, Pod},
+    util::get_type_name,
 };
+use anyhow::Result;
 use colored::Colorize;
-use glob::{GlobError, Paths};
 use regex::Regex;
+use serde::{de::DeserializeOwned, Serialize};
 use std::{
     collections::BTreeMap,
-    error::Error,
     fs,
-    iter::Map,
     path::{Path, PathBuf},
 };
 
+use super::{ItemKey, Store};
+
 #[derive(Debug)]
 pub struct LocalFileStore {
-    pub directory: PathBuf,
+    directory: PathBuf,
 }
 
 impl Store for LocalFileStore {
-    fn save_pod(&self, pod: &Pod) -> Result<(), Box<dyn Error>> {
-        let class = "pod";
+    fn save_pod(&self, pod: &Pod) -> Result<()> {
+        self.save_item(pod, &pod.hash, pod.annotation.as_ref())
+    }
 
-        // Save the annotation file and throw and error if exist
-        Self::save_file(
-            &self.make_annotation_path(
-                class,
-                &pod.hash,
-                &pod.annotation.name,
-                &pod.annotation.version,
-            ),
-            &serde_yaml::to_string(&pod.annotation)?,
-            true,
-        )?;
+    fn load_pod(&self, item_key: &ItemKey) -> Result<Pod> {
+        self.load_item::<Pod>(item_key)
+    }
 
-        // Save the pod and skip if it already exist, for the case of many annotation to a single pod
-        Self::save_file(
-            &self.make_spec_path(class, &pod.hash),
-            &to_yaml::<Pod>(pod)?,
-            false,
-        )?;
+    /// Return the name version index btree where key is name-version and value is hash of pod
+    fn list_pod(&self) -> Result<BTreeMap<String, String>> {
```

Comment:
One problem with this is the vectors will be locked a specific type such as String. I assume we are okay with that? A vector of struct would be more flexiable in the the future instead of a btreemap

Reply:
Chhange to vector of struct